Remove namelist spec_bdy_final_mu

weiwangncar opened the pull request:
Remove namelist spec_bdy_final_mu.

TYPE: no impact

KEYWORDS: lateral boundary call, spec_bdy_final, spec_bdy_final_mu

SOURCE: internal

DESCRIPTION OF CHANGES:
Problem:
The namelist spec_bdy_final_mu was introduced in 3.7 for an ad-hoc fix to the restart problem when spec_bdy_final calls were introduced to ensure the lateral boundary conditions do not drift by using tendencies only. The default has been to call this routine all the time, and not calling this routine does not fix any restart problem any more.

Solution:
Remove this namelist in the code, and make the code cleaner.

LIST OF MODIFIED FILES: 
M       Registry/Registry.EM_COMMON
M       dyn_em/solve_em.F
M       run/README.namelist
M       share/output_wrf.F
M       wrftladj/solve_em_ad.F
M       wrftladj/solve_em_tl.F

TESTS CONDUCTED: 
1. The code change does not have impact to the results.
2. The Jenkins tests all passing.

RELEASE NOTE: Namelist variable spec_bdy_final_mu is removed. It is redundant.

weiwangncar commented:
It passed the Jenkins tests:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 19           18
    Number of Builds       : 48           46
    Number of Simulations  : 163           161        0
    Number of Comparisons  : 103           102        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

davegill reviewed (approved):
Approved

dudhia commented:
Are you saying it has no use and there is still a restart problem after removing this?

weiwangncar commented:
@dudhia I'm saying that the code checks if spec_bdy_final_mu = 1, and the default value of this namelist is 1, so it doesn't do anything. Besides, at least for recent versions, setting this namelist to zero (meaning not calling spec_bdy_final for mu) does not do anything for the restart problem either.

dudhia commented:
Is this the smooth_opt restart problem or is there another?

On Mon, Apr 26, 2021 at 9:32 AM weiwangncar ***@***.***>
wrote:

> @dudhia <https://github.com/dudhia> I'm saying that the code checks if
> spec_bdy_final_mu = 1, and the default value of this namelist is 1, so it
> doesn't do anything. Besides, at least for recent versions, setting this
> namelist to zero (meaning not calling spec_bdy_final for mu) does not do
> anything for the restart problem either.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/wrf-model/WRF/pull/1490#issuecomment-826933615>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEIZ77AOVWQ7HXENFEK6SILTKWBPVANCNFSM43QOV73Q>
> .
>

weiwangncar commented:
@dudhia This PR is just code clean up, and it is not related to any restart problem, or correction of the restart problem.

weiwangncar commented:
@dudhia The namelist was introduced back in 3.7 to fix the restart problem we saw then. But it does not fix restart issue in the current code anymore.

weiwangncar commented:
@dudhia When we added the calls to spec_bdy_final in 3.7, we noticed nest restart problem. It was identified then that this particular call was the problem. So we added this namelist to not call this routine for MU so that nest could produce bit-for-bit restart. I don't know what smooth_option we were using at the time to test nest restart.

weiwangncar commented:
@dudhia Just tested: setting spec_bdy_final_mu = 0 does not help with restart issue associated with smooth_option = 2.

weiwangncar commented:
@dudhia Are you ok with this PR now?

dudhia reviewed (approved):
Looks fine
